tests/test_time_array.py

```diff
@@ -70,6 +74,23 @@ def test_radd(self):
         assert isinstance(x, ConstantTimeArray)
         assert_equal(x(0.0), [2, 3])
 
+    def test_batching(self):
+        # generic arrays
+        a = destroy(4)
+        H0 = a + dag(a)
+        psi0 = basis(4, 0)
+        times = jnp.linspace(0.0, 1.0, 11)
+
+        # constant time array
+        H_cte = jnp.stack([H0, 2 * H0])
+
+        # test sesolve output shape
+        result = sesolve(H_cte, psi0, times)
+        assert result.states.shape == (2, 11, 4, 1)
+
+        result = sesolve(H0 + H_cte, psi0, times)
+        assert result.states.shape == (2, 11, 4, 1)
+
```

Comment:
We should not use `sesolve` to test `TimeArray`, the unit tests of a file should not use other very high level function (every time we will break sesolve, we will also break these tests). If you want to add tests for this PR we should add them smwh in sesolve tests.

Reply:
Right, but `sesolve` tests call `closed_system` stuff right now, and this wouldn't fit in. How should I write that?

Reply:
For example in `tests/sesolve/test_sesolve_batching.py`. But it's a bit had-oc, basically this means that our vectorizing strategy is too tight with our solvers API, we should be able to test them independently. Doesn't matter much for now, just move this test smwh in `tests/sesolve`. 😉